pub-critic/src/app/components/ReviewForm/ReviewForm.tsx

```diff
@@ -0,0 +1,166 @@
+//wont split this into more components since it does not seem reusable, most likely will also just use default form handling since it only contains two fields and a star
+//form is custom (stars rating) so cannot use default form handling
+"use client";
+
+import React, { useState } from "react";
+import classes from "./ReviewForm.module.scss";
+import Image from "next/image";
+import star from "assets/star.svg";
+import lackOfStar from "assets/lackOfStar.svg";
+import { postReview, updateReview } from "api/ReviewsApi";
+
+interface ReviewFormProps {
+  gameId: number;
+  isEdit?: boolean;
+  initReview?: {
+    score: number;
+    title: string;
+    body: string;
+    id: number;
+  };
+  gameName: string;
+}
+
+export const ReviewForm = ({
+  gameId,
+  gameName,
+  initReview,
+  isEdit,
+}: ReviewFormProps) => {
+  const [rating, setRating] = useState<number>(
+    initReview ? initReview.score : 0
+  );
+  const [title, setTitle] = useState<string>(
+    initReview ? initReview.title : ""
+  );
+  const [body, setBody] = useState<string>(initReview ? initReview.body : "");
+  const [error, setError] = useState<string>("");
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    console.log(rating, title, body);
+    console.log("submitted");
+
+    if (rating === 0) {
+      setError("Please select a rating");
+      return;
+    }
+
+    if (title.length < 3) {
+      setError("Title must be at least 3 characters long");
+      return;
+    }
+
+    if (body.length < 10) {
+      setError("Body must be at least 10 characters long");
+      return;
+    }
+
+    if (isEdit && body === initReview!.body && title === initReview!.title) {
+      setError("You must change something to update your review");
+      return;
+    }
+
+    const response = isEdit
+      ? await updateReview(initReview!.id, {
```

Comment:
Why is .id optional? 🤔

Reply:
what exactly do you mean by optional? This is meant to update an existing review or create a new one, so the id is provided only in the case of an update